transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,146 @@
 
 if TYPE_CHECKING:
-    from MDAnalysis.core.universe import Universe, AtomGroup
+    from MDAnalysis.core.universe import AtomGroup
 
 
 class VelocityAutocorr(AnalysisBase):
-    """VelocityAutocorr class.
-
-    This class is used to perform analysis on a trajectory.
+    """
+    Class to calculate a velocity autocorrelation function (VACF).
 
     Parameters
     ----------
-    universe_or_atomgroup: :class:`~MDAnalysis.core.universe.Universe` or :class:`~MDAnalysis.core.groups.AtomGroup`
-        Universe or group of atoms to apply this analysis to.
-        If a trajectory is associated with the atoms,
-        then the computation iterates over the trajectory.
-    select: str
-        Selection string for atoms to extract from the input Universe or
-        AtomGroup
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+        Desired dimensions to be included in the VACF. Defaults to 'xyz'.
+    fft : bool
+        If ``True``, uses a fast FFT based algorithm for computation of
+        the VACF. Otherwise, use the simple "windowed" algorithm.
+        The tidynamics package is required for `fft=True`.
+        Defaults to ``True``.
 
     Attributes
     ----------
-    universe: :class:`~MDAnalysis.core.universe.Universe`
-        The universe to which this analysis is applied
-    atomgroup: :class:`~MDAnalysis.core.groups.AtomGroup`
+    atomgroup : :class:`~MDAnalysis.core.groups.AtomGroup`
         The atoms to which this analysis is applied
-    results: :class:`~MDAnalysis.analysis.base.Results`
-        results of calculation are stored here, after calling
-        :meth:`VelocityAutocorr.run`
-    start: Optional[int]
-        The first frame of the trajectory used to compute the analysis
-    stop: Optional[int]
-        The frame to stop at for the analysis
-    step: Optional[int]
-        Number of frames to skip between each analyzed frame
-    n_frames: int
-        Number of frames analysed in the trajectory
-    times: numpy.ndarray
+    dim_fac : int
+        Dimensionality :math:`d` of the VACF.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged VACF over all the particles with respect to lag-time.
+        Obtained after calling :meth:`VelocityAutocorr.run`
+    results.vacf_by_particle : :class:`numpy.ndarray`
+        The VACF of each individual particle with respect to lag-time.
+    start : Optional[int]
+        The first frame of the trajectory used to compute the analysis.
+    stop : Optional[int]
+        The frame to stop at for the analysis.
+    step : Optional[int]
+        Number of frames to skip between each analyzed frame.
+    n_frames : int
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles VACF was calculated over.
+    times : numpy.ndarray
```

Comment:
Docstring? / what is `times` going to do?

Reply:
`times` likely will not be used explicitly in this class. Would it be best to remove it from the docstring then?

Reply:
Probably.